examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -58,100 +57,363 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
 }
 """
 function Initialise_DYCOMS!(state::Vars, aux::Vars, (x,y,z), t)
-  DT         = eltype(state)
-  xvert::DT  = z
-  
-  epsdv::DT     = molmass_ratio
-  q_tot_sfc::DT = 8.1e-3
-  Rm_sfc::DT    = gas_constant_air(PhasePartition(q_tot_sfc))
-  ρ_sfc::DT     = 1.22
-  P_sfc::DT     = 1.0178e5
-  T_BL::DT      = 285.0
-  T_sfc::DT     = P_sfc/(ρ_sfc * Rm_sfc);
-  
-  q_liq::DT      = 0
-  q_ice::DT      = 0
-  zb::DT         = 600   
-  zi::DT         = 840 
+  FT            = eltype(state)
+  xvert::FT     = z
+  #These constants are those used by Stevens et al. (2005)
+  qref::FT      = 7.75e-3
+  q_tot_sfc::FT = qref
+  q_pt_sfc      = PhasePartition(q_tot_sfc)
+  Rm_sfc        = gas_constant_air(q_pt_sfc)
+  T_sfc::FT     = 292.5
+  P_sfc::FT     = MSLP
+  ρ_sfc::FT     = P_sfc / Rm_sfc / T_sfc
+  # Specify moisture profiles 
+  q_liq::FT      = 0
+  q_ice::FT      = 0
+  zb::FT         = 600    # initial cloud bottom
+  zi::FT         = 840    # initial cloud top
   dz_cloud       = zi - zb
-  q_liq_peak::DT = 4.5e-4
-  
+  q_liq_peak::FT = 0.00045 #cloud mixing ratio at z_i    
   if xvert > zb && xvert <= zi        
     q_liq = (xvert - zb)*q_liq_peak/dz_cloud
   end
-  if ( xvert <= zi)
-    θ_liq  = DT(289)
-    q_tot  = DT(8.1e-3)
+  if xvert <= zi
+    θ_liq = FT(289)
+    q_tot = qref
   else
-    θ_liq = DT(297.5) + (xvert - zi)^(DT(1/3))
-    q_tot = DT(1.5e-3)
+    θ_liq = FT(297.5) + (xvert - zi)^(FT(1/3))
+    q_tot = FT(1.5e-3)
   end
-
-  q_pt = PhasePartition(q_tot, q_liq, DT(0))
-  Rm    = gas_constant_air(q_pt)
-  cpm   = cp_m(q_pt)
+  # Calculate PhasePartition object for vertical domain extent
+  q_pt  = PhasePartition(q_tot, q_liq, q_ice) 
   #Pressure
-  H = Rm_sfc * T_BL / grav;
-  P = P_sfc * exp(-xvert/H);
-  #Exner
-  exner_dry = exner(P, PhasePartition(DT(0)))
-  #Temperature 
-  T             = exner_dry*θ_liq + LH_v0*q_liq/(cpm*exner_dry);
-  #Density
-  ρ             = P/(Rm*T);
-  #Potential Temperature
-  θv     = virtual_pottemp(T, P, q_pt)
-  # energy definitions
-  u, v, w     = DT(7), DT(-5.5), DT(0)
-  U           = ρ * u
-  V           = ρ * v
-  W           = ρ * w
-  e_kin       = DT(1//2) * (u^2 + v^2 + w^2)
+  H     = Rm_sfc * T_sfc / grav;
+  p     = P_sfc * exp(-xvert/H);
+  #Density, Temperature
+  TS    = LiquidIcePotTempSHumEquil_no_ρ(θ_liq, q_pt, p)
+  ρ     = air_density(TS)
+  T     = air_temperature(TS)
+  #Assign State Variables
+  u, v, w     = FT(7), FT(-5.5), FT(0)
+  e_kin       = FT(1/2) * (u^2 + v^2 + w^2)
   e_pot       = grav * xvert
   E           = ρ * total_energy(e_kin, e_pot, T, q_pt)
   state.ρ     = ρ
-  state.ρu    = SVector(U, V, W) 
+  state.ρu    = SVector(ρ*u, ρ*v, ρ*w) 
   state.ρe    = E
   state.moisture.ρq_tot = ρ * q_tot
-end   
+end
+
+
+# TODO: temporary; move to new CLIMA module
+# TODO: add an option to reduce communication: compute averages
+# locally only
+function gather_diagnostics(dg, Q)
```

Comment:
Now that the code has been tested, it would be good to move this function somewhere to be used by any driver.

Reply:
Yes, that's the plan. Will work that out this week, likely.